app.js

```diff
@@ -15,5 +15,50 @@ app.get('/', (req, res) => {
   res.send('oh hello')
 });
 
+app.get('/api/v1/projects/', async(request, response) => {
+  try {
+    const project = await database('projects').select();
+    response.status(200).json(project)
+  } catch (error) {
+    response.status(500).json({error: 'internal server error' })
+  }
+});
+
+app.get('/api/v1/projects/:id', async(request, response) => {
+  const { id } = request.params;
+  try {
+    const project = await database('projects').where('id', id);
+
+    if (project.length) {
+      response.status(200).json(project)
+    } else {
+      response.status(404).json({
+        error: `Could not find a project with id: ${request.params.id}`
+      });
+    }
+  } catch (error) {
+    response.status(500).json({error: 'internal server error' })
+  }
+});
+
+app.post('/api/v1/projects', async (request, response) => {
+  const project = request.body;
+
+  for (let requiredParameter of ['title']) {
+    if (!project.hasOwnProperty(requiredParameter)) {
```

Comment:
Even though only one parameter of `title` is being passed, this is great, as it allows for scalability. In the future, say, we need/have to add more params, this allows that to take place easily. Good work